### PR TITLE
Adding table prefix handling

### PR DESCRIPTION
**Pull request by burzum**
Replaces https://github.com/cakephp/bake/pull/191

### CODE REVIEW

File: src/Shell/Task/ModelTask.php [hunk 1]
```diff
@@ -317,11 +317,13 @@ public function findTableReferencedBy($schema, $keyField)
                 if (!isset($constraintInfo['references'])) {
                     continue;
                 }
-
+                $length = mb_strlen($this->tablePrefix);
+                if ($length > 0 && substr($constraintInfo['references'][0], 0, $length) === $this->tablePrefix) {
```
Comment (dereuromark) [inline src/Shell/Task/ModelTask.php]: mb_substr if you used mb_ above.

File: src/Shell/Task/BakeTask.php [hunk 2]
```diff
@@ -30,6 +30,13 @@ class BakeTask extends Shell
     use ConventionsTrait;
 
     /**
+     * Table prefix
+     *
+     * @var string
```
Comment (dereuromark) [inline src/Shell/Task/BakeTask.php]: |null

**Comment by dereuromark** [comment]
Apart from my comments this is a very useful feature good to go from my end 👍

**Comment by dereuromark** [comment]
@burzum Are you able to come up with a small test here? I love the feature and I am sure many will like having the possibility of setting a custom table prefix.

**Comment by burzum** [comment]
@dereuromark not right now, just add the test if you're in a hurry. Meanwhile all our prefix stuff has been more or less resolved so I'm not in an urgent need of that feature anymore.

**Comment by dereuromark** [comment]
PR with test opened.

**Pull request closed without merging**